spring-security-oauth/src/main/java/org/springframework/security/oauth/provider/filter/CoreOAuthProviderSupport.java

```diff
@@ -1,268 +1,310 @@
+						String value = oauthDecode(entry.getValue());
+						parsedEntries.put(key, value);
+					}
+					catch (DecoderException e) {
+						throw new IllegalStateException(e);
+					}
+				}
+			}
 
-      //create a map of the authorization header values per OAuth Core 1.0, section 5.4.1
-      String[] headerEntries = StringSplitUtils.splitIgnoringQuotes(authHeaderValue, ',');
-      for (Object o : StringSplitUtils.splitEachArrayElementAndCreateMap(headerEntries, "=", "\"").entrySet()) {
-        Map.Entry entry = (Map.Entry) o;
-        try {
-          String key = oauthDecode((String) entry.getKey());
-          String value = oauthDecode((String) entry.getValue());
-          parameters.put(key, value);
-        }
-        catch (DecoderException e) {
-          throw new IllegalStateException(e);
-        }
-      }
-    }
+			parameters = new OAuthParameters();
+			parameters.setConsumerKey(parsedEntries.get(OAuthParameter.oauth_consumer_key.toString()));
+			parameters.setToken(parsedEntries.get(OAuthParameter.oauth_token.toString()));
+			parameters.setTokenSecret(parsedEntries.get(OAuthParameter.oauth_token_secret.toString()));
+			parameters.setSignatureMethod(parsedEntries.get(OAuthParameter.oauth_signature_method.toString()));
+			parameters.setSignature(parsedEntries.get(OAuthParameter.oauth_signature.toString()));
+			parameters.setTimestamp(parsedEntries.get(OAuthParameter.oauth_timestamp.toString()));
+			parameters.setNonce(parsedEntries.get(OAuthParameter.oauth_nonce.toString()));
+			parameters.setVersion(parsedEntries.get(OAuthParameter.oauth_version.toString()));
+			parameters.setCallback(parsedEntries.get(OAuthParameter.oauth_callback.toString()));
+			parameters.setCallbackConfirmed(parsedEntries.get(OAuthParameter.oauth_callback_confirmed.toString()));
+			parameters.setVerifier(parsedEntries.get(OAuthParameter.oauth_verifier.toString()));
+			parameters.setRealm(parsedEntries.get("realm"));
+		}
 
-    return parameters;
-  }
+		return parameters;
+	}
 
-  /**
-   * Get the supported OAuth parameters. The default implementation supports only the OAuth core parameters.
-   *
-   * @return The OAuth core parameters.
-   */
-  protected Set<String> getSupportedOAuthParameters() {
-    return this.supportedOAuthParameters;
-  }
+	@Override
+	public String getSignatureBaseString(HttpServletRequest request) {
+		SortedMap<String, SortedSet<String>> significantParameters = loadSignificantParametersForSignatureBaseString(request);
 
-  // Inherited.
-  public String getSignatureBaseString(HttpServletRequest request) {
-    SortedMap<String, SortedSet<String>> significantParameters = loadSignificantParametersForSignatureBaseString(request);
+		// now concatenate them into a single query string according to the spec.
+		StringBuilder queryString = new StringBuilder();
+		Iterator<Map.Entry<String, SortedSet<String>>> paramIt = significantParameters.entrySet().iterator();
+		while (paramIt.hasNext()) {
```

Comment:
Perhaps because it is easier to check whether we need to add '&':
```java
				if (paramIt.hasNext() || valueIt.hasNext()) {
					queryString.append('&');
				}
```
Alternative with "for each" would be to add '&' anyway, then check and delete after the end of the loop.